MRML/vtkIGTLToMRMLPolyData.cxx

```diff
@@ -380,10 +380,9 @@ int vtkIGTLToMRMLPolyData::MRMLToIGTL(unsigned long event, vtkMRMLNode* mrmlNode
     
     //------------------------------------------------------------
     // Allocate Status Message Class
-    if (this->OutPolyDataMessage.IsNull())
-      {
-      this->OutPolyDataMessage = igtl::PolyDataMessage::New();
-      }
+    // Overwrite old message because we want to clear arrays from previous
+    // calls that will not be set anew.
+    this->OutPolyDataMessage = igtl::PolyDataMessage::New();
```

Comment:
After looking at igtl::PolyDataMessage:Clear(), it seems like the method does something else than it says. There are plenty of Clear() methods in the PolyDataPointArray/CellArray/Attribute-classes, but they are not used by  igtl::PolyDataMessage:Clear(). Thus I agree with @lassoan .

I can 
* Withdraw this pull request.
* Create a new pull request containing all the other proposed changed, excluding the debated line.
* Implement option A from @lassoan and then in vtkIGTLToMRMLPolyData::MRMLToIGTL: call igtl::PolyDataMessage:Clear() before reusing the message object. This will replace my original proposal.

This will take a few days as I must find my Slicer-test setup somewhere on my other Mac.

Although I agree on your other suggestions (Squeeze() and igtl::NDArrayMessage..), I dont have a use case for these and thus limited ability to test them.